improver/spotdata/spot_extraction.py

```diff
@@ -92,7 +92,11 @@ def extract_coordinates(self, neighbour_cube):
         index_constraint = iris.Constraint(grid_attributes_key=["x_index", "y_index"])
```

Comment:
I've noticed that the spot extraction uses x-y coord order for slicing gridded input, which is the opposite to what we enforce on load. @bayliffe, is there any particular reason for it?

Reply:
Only that my brain prefers x-y ordering. I noticed this too when reviewing this. Happy for it to be reversed.